SEED-platform/seed
# Handle type casting in data quality checks

#### Any background context you want to provide?
The typecasting within the data quality checks were not handling the case where a string was defined as a number and has a min/max. This caused the data quality checks to fail ungracefully preventing future checks to occur

#### What's this PR do?
* Update the minimum_valid and maximum_valid checks to correctly handle an invalid typecasting call.
* Add tests

#### How should this be manually tested?
Test 1:
* Modify the data quality rule of a text non-extra-data field (e.g. Address Line 1) so that the rule data type is set to Number, Date, Year, Area, or EUI and it has a minimum and/or maximum value set
* Upload the file and proceed to the Mapping Review screen where data quality results are loaded

Test 2:
* Set an EUI min and max with the unit being either Number or EUI.
* Verify that the DQ's run and report issues appropriately.

Test 3:
* Set the units of the site EUI field to feet squared. This makes no sense, but SEED needs to handle this nicely
* Run a DQ on a record and verify that the DQ results in a warning about mismatched dimensions.

#### What are the relevant tickets?
#1817 
#1845 
#1873 
#1375 (alex)

#### Screenshots (if appropriate)
